Contributing.md

alepev opened the pull request:
Closes #191 

Includes conventions discussed in https://github.com/thehyve/delphyne-template/pull/73

Related to https://github.com/thehyve/delphyne-template/pull/76

alepev commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,194 @@
+Once your changes are ready to be submitted, push them to your fork of the GitHub repo before creating a pull request. 
+Depending on the type of issue the pull request is resolving, it should merge onto the appropriate branch:
+
+- pull requests to delphyne: merge to `release/X.X.X` branch if specified in the issue milestone, otherwise `master`.
+- pull request to delphyne-template: merge to `release/X.X.X` branch if specified in the issue milestone, otherwise `dev`.
```
do we know how to use issue milestones? could be useful..

alepev commented:
To see the result: https://github.com/thehyve/delphyne/blob/contributing/CONTRIBUTING.md

Spayralbe commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,194 @@
+
+Both delphyne and delphyne-template make use of [semantic versioning](https://semver.org/).
+
+#### delphyne releases
```
Mention somewhere here that a new release of delphyne should be reflected in the requirements.txt file of delphyne-template

alepev replied:
done

Spayralbe reviewed (commented):
Clear and thorough ✨

alepev commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,194 @@
+
+### Getting started
+
+To contribute to the project, you should be already familiar with [git](https://git-scm.com/).
```
agreed

alepev commented on CONTRIBUTING.md:
```diff
@@ -1,23 +1,194 @@
+
+## Contributing as a developer
+
+You should always work on delphyne and [delphyne-template](https://github.com/thehyve/delphyne-template) side-by-side.
+It is your responsibility as a developer to maintain compatibility between the two projects,
+making sure that changes to one repository are reflected by complementary changes in the other.
```
Yes it came across as a bit harsh 😅 like "don't mess with my repo you punks!"

alepev commented:
- addressed comments
- changed titles & text to make clear the instructions are not just for delphyne, but also delphyne-template
- fixed couple links
- other
